docs/developer-guide/running-substreams.md

```diff
@@ -77,3 +77,36 @@ map_transfers: message "eth.erc721.v1.Transfers": {
 ```
 
 The example output shown above contains data for different transfers from data in the blockchain. These transfers can also be verified on [Etherscan](https://etherscan.io/tx/0xcfb197f62ec5c7f0e71a11ec0c4a0e394a3aa41db5386e85526f86c84b3f2796).
+
+## Development & Production Mode
+
+### Development Mode
+
+Production and development modes impact the execution of Substreams. Key aspects of execution include:
+
+- The time required to reach the first byte.
+- The speed that large ranges get executed.
+- The module logs and outputs that are sent back to the client.
+
+Differences between production and development modes include:
+
+- Forward execution is enabled in production mode and disabled in development mode.
+- The time required to reach the first byte in development mode is faster than in production mode.
+- In development mode, the client will receive all of the executed module's logs. In production mode the client will only receive the output module's logs.
+- In development mode, it's possible to request specific store snapshots in the execution tree.
+
+Examples: (given the dependencies: `[block] --> [map_pools] --> [store_pools] --> [map_transfers])`
+
+- Running the `substreams run substreams.yaml map_transfers` command will only print the `map_transfers` module's outputs and logs.
+- Running the `substreams run substreams.yaml map_transfers --debug-modules-output=map_pools,map_transfers,store_pools` command will print the outputs of the `map_pools`, `map_transfers`, and `store_pools` modules.
+- Running the `substreams run substreams.yaml map_transfers -s 1000 -t +5 --debug-modules-initial-snapshot=store_pools` command will print all the entries in the `store_pools` module at block 999, then continue with outputs and logs from the `map_transfers` module in blocks 1000 through 1004.
+
+### Parallel Execution for Substreams Modes
+
+The two steps involved during parallel execution are backward execution and forward execution.
+
+Backward execution consists of executing in parallel block ranges, from the module's initial block, up to the start block of the request. If the start block of the request matches the module's initial block no backwards execution is performed.
+
+Forward execution consists of executing in parallel block ranges from the start block of the request up to last known final block, also called an irreversible block, or the stop block of the request depending on which is smaller. Forward execution significantly improves the performance of Substreams, however the ability to stream module logs is lost.
```

Comment:
We should not mention that fact that module's log are lost, it would be technically feasible even in forward parallel execution but we decided not. The fact that there is no logs should be only mention because of `production` vs `development`.

Reply:
Updated to remove the wording referencing any lost logs. The changes will be available in the next commit/push.